asm/unknown_070110/func_800708D0.s

```diff
@@ -0,0 +1,94 @@
+/* 071518 80070918 008C2021 */  addu  $a0, $a0, $t4
+/* 07151C 8007091C 00000000 */  nop   
+/* 071520 80070920 004D0019 */  multu $v0, $t5
+/* 071524 80070924 00421026 */  xor   $v0, $v0, $v0
+/* 071528 80070928 00006812 */  mflo  $t5
+/* 07152C 8007092C 00AD2821 */  addu  $a1, $a1, $t5
+/* 071530 80070930 00000000 */  nop   
+.L80070934:
+/* 071534 80070934 94CD0002 */  lhu   $t5, 2($a2)
+/* 071538 80070938 94CC000E */  lhu   $t4, 0xe($a2)
+/* 07153C 8007093C 24C6000C */  addiu $a2, $a2, 0xc
+/* 071540 80070940 018D6023 */  subu  $t4, $t4, $t5
+.L80070944:
+/* 071544 80070944 84AD0000 */  lh    $t5, ($a1)
+/* 071548 80070948 84AE0002 */  lh    $t6, 2($a1)
+/* 07154C 8007094C 84AF0004 */  lh    $t7, 4($a1)
+/* 071550 80070950 01A90018 */  mult  $t5, $t1
+/* 071554 80070954 90B80007 */  lbu   $t8, 7($a1)
+/* 071558 80070958 90A70006 */  lbu   $a3, 6($a1)
+/* 07155C 8007095C 90B90008 */  lbu   $t9, 8($a1)
+/* 071560 80070960 0018C400 */  sll   $t8, $t8, 0x10
+/* 071564 80070964 90A30009 */  lbu   $v1, 9($a1)
+/* 071568 80070968 00F8C025 */  or    $t8, $a3, $t8
+/* 07156C 8007096C 00006812 */  mflo  $t5
+/* 071570 80070970 00000000 */  nop   
+/* 071574 80070974 00000000 */  nop   
+/* 071578 80070978 01CA0018 */  mult  $t6, $t2
+/* 07157C 8007097C 00007012 */  mflo  $t6
+/* 071580 80070980 01AE6820 */  add   $t5, $t5, $t6
+/* 071584 80070984 00000000 */  nop   
+/* 071588 80070988 01EB0018 */  mult  $t7, $t3
+/* 07158C 8007098C 00007812 */  mflo  $t7
+/* 071590 80070990 01AF6820 */  add   $t5, $t5, $t7
+/* 071594 80070994 19A00007 */  blez  $t5, .L800709B4
+/* 071598 80070998 00000000 */   nop   
+/* 07159C 8007099C 000D6D82 */  srl   $t5, $t5, 0x16
+/* 0715A0 800709A0 006D1821 */  addu  $v1, $v1, $t5
+/* 0715A4 800709A4 2C6D0081 */  sltiu $t5, $v1, 0x81
+/* 0715A8 800709A8 15A00002 */  bnez  $t5, .L800709B4
+/* 0715AC 800709AC 00000000 */   nop   
+/* 0715B0 800709B0 34030080 */  ori   $v1, $zero, 128
+.L800709B4:
+/* 0715B4 800709B4 00780019 */  multu $v1, $t8
+/* 0715B8 800709B8 24A5000A */  addiu $a1, $a1, 0xa
+/* 0715BC 800709BC 258CFFFF */  addiu $t4, $t4, -1
+/* 0715C0 800709C0 2484000A */  addiu $a0, $a0, 0xa
+/* 0715C4 800709C4 0000C012 */  mflo  $t8
+/* 0715C8 800709C8 0018C1C2 */  srl   $t8, $t8, 7
+/* 0715CC 800709CC A098FFFC */  sb    $t8, -4($a0)
+/* 0715D0 800709D0 00790019 */  multu $v1, $t9
+/* 0715D4 800709D4 0018C402 */  srl   $t8, $t8, 0x10
+/* 0715D8 800709D8 A098FFFD */  sb    $t8, -3($a0)
+/* 0715DC 800709DC 0000C812 */  mflo  $t9
+/* 0715E0 800709E0 0019C9C2 */  srl   $t9, $t9, 7
+/* 0715E4 800709E4 A099FFFE */  sb    $t9, -2($a0)
+/* 0715E8 800709E8 5580FFD6 */  bnezl $t4, .L80070944
+/* 0715EC 800709EC 00000000 */   nop   
+/* 0715F0 800709F0 2508FFFF */  addiu $t0, $t0, -1
+/* 0715F4 800709F4 5500FFBE */  bnezl $t0, .L800708F0
+/* 0715F8 800709F8 00000000 */   nop   
+/* 0715FC 800709FC 03E00008 */  jr    $ra
+/* 071600 80070A00 00000000 */   nop   
+
+/* 071604 80070A04 94CC0002 */  lhu   $t4, 2($a2)
```

Comment:
Make this a standalone function?

Reply:
Hmm, it's a bit of an odd one. Lends credence to it's most likely state as a hand written chunk of assembly though. On line 85 of that file it has the label: `L80070A08` which is jumped to from the line 13 in `func_800708D0`. But yeah, that line 84 lhu suggests that there's a function there that is just unused that would start on that instruction and then lead into the next ones on it's own, and then oddly jump back into the middle of `func_800708D0` from line 91. So this is just a weird blend of functions that can jump in and out of each other in a way that only hand written assembly could do?

Reply:
Ah, missed the jump into this from above. Leaving together sgtm, thanks.